api_spec.yaml

```diff
@@ -1272,6 +1477,125 @@ paths:
+                                              default: "PHONE_NUMBER_ALREADY_EXISTS_ERROR"
+                                    - type: object
+                                      properties:
+                                          status:
+                                              type: string
+                                              default: "THIRD_PARTY_USER_ALREADY_EXISTS_ERROR"
+                                    - type: object
+                                      properties:
+                                          status:
+                                              type: string
+                                              default: "ASSOCIATION_NOT_ALLOWED_ERROR"
+                                          reason:
+                                              type: string
+
+                401:
+                    description: Unauthorised access
+                    content:
+                        text/plain:
+                            schema:
+                                type: string
+                                enum:
+                                    - Unauthorised access
+    /dashboard/api/tenants/user/disassociate:
+        put:
+            tags:
+                - Multitenancy
+            summary: Disassociate a user from a tenant
+            parameters:
+                - name: authorization
+                  in: header
+                  required: true
+                  schema:
+                      type: string
+                      example: "Bearer API_KEY"
+            requestBody:
+                content:
+                    application/json:
+                        schema:
+                            type: object
+                            properties:
+                                tenantId:
+                                    type: string
+                                userId:
+                                    type: string
+            responses:
+                200:
+                    description: Success
+                    content:
+                        application/json:
+                            schema:
+                                type: object
+                                properties:
+                                    status:
+                                        type: string
+                                        default: "OK"
+                                    wasAssociated:
+                                        type: boolean
+                                        default: true
+
+                401:
+                    description: Unauthorised access
+                    content:
+                        text/plain:
+                            schema:
+                                type: string
+                                enum:
+                                    - Unauthorised access
     /dashboard/api/tenants/login-methods:
         get:
             tags:
```

Comment:
Also add the API for boxy SAML upload

Reply:
Should we add this API to the backend SDK since the main request to get the client id and secret via XML happens via boxy for which the URL would be provided by the user themselves, should we not just call that API on the client side directly?

Reply:
No. I think you should add the API to the dashboard only. this API will inturn call the boxy url. The response contains client id and secret. You must search if the matching client id already exists, and update it if it does. Else add it to the clients array